src/runtime_config/operator_gov.lua

```diff
@@ -0,0 +1,130 @@
+-- Copyright (c) Microsoft Corporation. All rights reserved.
+-- Licensed under the Apache 2.0 License.
+
+-- This file defines the default initial contents (ie, Lua scripts) of the gov_scipts table.
+return {  
+  pass = [[
+  tables, calls, votes = ...
+
+  -- defines which of the members are operators
+  function is_operator(member)
+    return member == "0"
+  end
+
+  -- defines calls that can be passed with sole operator input
+  operator_calls = {
+    accept_node=true,
+    retire_node=true,
+    new_code=true
+  }
+
+  operator_votes = 0
+  member_votes = 0
+
+  for member, vote in pairs(votes) do
+    if vote then
+      if is_operator(member) then
+        operator_votes = operator_votes + 1
+      else
+        member_votes = member_votes + 1
+      end
+    end
+  end
+
+  -- count active members, excluding operators
+  members_active = 0
+  STATE_ACTIVE = 1
+
+  tables["ccf.members"]:foreach(function(member, details) 
+    if details["status"] == STATE_ACTIVE and not is_operator(tostring(member)) then
+      members_active = members_active + 1 
+    end 
+  end)
+
+  -- check for raw_puts to sensitive tables
+  SENSITIVE_TABLES = {"ccf.whitelists", "ccf.gov_scripts"}
+  for _, call in pairs(calls) do
+    if call.func == "raw_puts" then
+      for _, sensitive_table in pairs(SENSITIVE_TABLES) do
+        if call.args[sensitive_table] then
+          -- require unanimity of non-operating members
+          return member_votes == members_active 
+        end
+      end
+    end
+  end
+
+  -- a vote is an operator vote if it's only making operator calls
+  operator_vote = true;
+  for _, call in pairs(calls) do
+    if not operator_calls[call.func] then
+      operator_vote = false
+      break
+    end
+  end
+
+  -- a majority of members can always pass votes
+  if member_votes > math.floor(members_active / 2) then
+    return true
+  end
+
+  -- a single operator can pass an operator vote
+  if operator_vote then
+    return operator_votes > 0
+  end
+
+  return false]],
+
+  environment_proposal = [[
```

Comment:
Do we need to define these environment proposals every time? Can we register a default one and override if it is defined?

Reply:
Every time? This is only defined once per constitution. We could register a default one, as could we with the constitution itself, but having all the constitution explicitly in one file also seems good from an audit perspective (users should read it at least once!). Users wouldn't want to pick up a new environment_proposal without refreshing their constitution accordingly (the two may not be compatible!), so if we start splitting them out, we have to worry about versions etc.

This seems to be equivalent to the discussion about whether we should have the simple and operator constitutions separate, or as one, parametrised version. If we split them, we have some duplication, but the simple one really is simple. If we don't, we don't have duplication, but people who go for simple have to audit a bunch of extra code they don't need.

Either way, we have to test both. We're also not heading towards having N samples, and we definitely don't want to worry about dependency management and packaging for Lua in the KV (at least not now!).

So in short, I think I'd rather not. But given compelling arguments, I could change my mind :)